src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -1207,7 +1256,9 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
           deleteRemainingChildren(returnFiber, child.sibling);
           const existing = useFiber(child, expirationTime);
           existing.ref = coerceRef(child, element);
-          existing.pendingProps = element.props;
+          existing.pendingProps = element.type === REACT_FRAGMENT_TYPE
+            ? element.props.children
```

Comment:
Unlike the other branches, this doesn't do the wrapper thing? (It's good that it doesn't but inconsistent. Probably indicates that there is a missing test.)

Reply:
This unifies both behind `child.type === element.type`. This is neat but it also means that we're sometimes comparing strings to numbers/symbols. That could potentially deopt the type hints in this function. It might actually be better to check the `child.tag` to ensure that it is a `HostComponent` or `Fragment` and compare the type in separate branches.

In fact, that might already be a bug incase we ever add strings to `fiber.type` on other fibers than HostComponent.